apps/web/src/components/Common/Layout.tsx

```diff
@@ -104,9 +106,11 @@ const Layout: FC<LayoutProps> = ({ children }) => {
       <GlobalAlerts />
       <div className="flex min-h-screen flex-col pb-14 md:pb-0">
         <Navbar />
-        <GlobalBanners />
-        <BottomNavigation />
-        {children}
+        <div className={cn({ 'pt-[5.8rem]': staffMode, 'pt-16': !staffMode })}>
```

Comment:
this will create huge huge huge problem, it wraps 100s of 1000s of DOM in this div and an good approach 🙇🏼 what caused this issue?

Reply:
We are adding a class to the div, it will not cause any types of re renders in IMO

Reply:
no rerenders, it adds extra dom to the root!

Reply:
https://www.commercegurus.com/dom-size/

Reply:
Yeah, I understand but how it can cause a huge huge problem?

Reply:
its a new div!

Reply:
@bigint Can we discuss it? I can explain you why that div is needed.

Reply:
@bigint This div is needed to create a separation between navbar and actual content of the page. It will increase the overall DOM size by just 1. I don't think that it will have any impact on the performance of the application.
There is no other way to handle this. The behaviour of radix components and headless components are quite different. May be we can discuss on discord, if you want more explaination.

Reply:
But question is why its affecting areas that are not using radix?

Reply:
@bigint Whenever a modal is opened in radix, we will need to make the document element overflow property to hidden to disable the parent scroll, because of which the navbar having sticky position will not work and will not be visible (if parent overflow is hidden, child descendents with sticky position will not work). that's why I have change the navbar position to fixed. Now the problem is that since the position of navbar is fixed, it will hide the content behind it, to solve that we will need to have a wrapper around the content and will need to add padding top to it which equals to the navbar height to make the content visible below navbar.

Reply:
where is the code affecting this?

Reply:
@bigint Please refer this useEffect.
https://github.com/heyxyz/hey/blob/58cf03abcc89265c7e3693e44b91c20331d4807f/packages/ui/src/Modal.tsx#L27

Reply:
Because of which the navbar sticky position will not work and it will be hided from the background, that's why changed navbar position to fixed. Refer
https://github.com/heyxyz/hey/blob/1aa3372b45ee28888ee6de03aae6073229c6e2a5/apps/web/src/components/Shared/Navbar/index.tsx#L72

Reply:
@bigint Can you pls help me to understand what you are worrying about in this change?

Reply:
@bigint Any Updates?

Reply:
@sdthakral33 i mean which primitive makes the UI looks ugly?

Reply:
im worried because adding a root level dom will increase the dom depth, which affects web vitals drastically

Reply:
![image](https://github.com/heyxyz/hey/assets/69431456/2b5df96d-2b5c-47be-b591-ad8683c0cd97)

it will add a another parent to the dom, i could have accepted if the dom is appended anywhere in the middle, but parent is huge!

Reply:
https://github.com/heyxyz/hey/assets/69431456/f992172f-61a0-4cd9-a699-2bafdf2a5b98

Reply:
@bigint If we want to use radix primitive then there is not other way, it will make the body overflow ahidden (part of radix framework), because of which we cannot use sticky navbar with radix. If we will be going to change the position of navbar to fixed, then we will need to add wrapper around the main content just after the navbar to adjust the content according to the fixed navbar.

Reply:
> ![image](https://private-user-images.githubusercontent.com/69431456/293379165-2b5df96d-2b5c-47be-b591-ad8683c0cd97.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM4NjAyNDUsIm5iZiI6MTcwMzg1OTk0NSwicGF0aCI6Ii82OTQzMTQ1Ni8yOTMzNzkxNjUtMmI1ZGY5NmQtMmI1Yy00N2JlLWI1OTEtYWQ4NjgzYzBjZDk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyMzEyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI5VDE0MjU0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRkODJmN2FiNDEyYTlkZjY4Njk2YTUzMzA0OTllYTg1NjFiZjAzYzhiYzExZGI1ZGU0NzJiNWM4M2IyMjQ3MDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.74yApq9x6lO8L1iqygBHG0BdyjCF7qPZJLu8QORe1PI)
> 
> it will add a another parent to the dom, i could have accepted if the dom is appended anywhere in the middle, but parent is huge!

Yeah, I understand that but it will increase the depth by just one. IMO that should not be a problem.

Reply:
If that div would have increased the depth by some n (large) number, then there can be web vitals issues.

Reply:
@bigint I can fix this. Just gave a thought about it.

Reply:
@bigint Fixed this in the latest commit. Please review!

Reply:
https://github.com/heyxyz/hey/pull/4347/files#diff-0e16eaa722dcbe43f850552755cc5fafac704a32a39761450c6d2edf5a91a637

Reply:
@bigint Any Updates? Is this fine?

Reply:
> Radix Primitive Dialog will make the UI look ugly.

Are they removing the style while without even consuming them? I don't see any of the platform using radix doing this to manually add multiple padding, this looks like a antipattern and not the fix itself 🙇🏼

Reply:
<img width="1512" alt="Screenshot 2023-12-30 at 1 09 10 PM (2)" src="https://github.com/heyxyz/hey/assets/68169950/a68a458a-29f8-4efc-ae00-7e5c42458f8c">
@bigint Radix UI itself uses padding top to create separation between header and main content. Please refer the above screenshot you can see `padding-top: var(--header-height)`

Reply:
but we already have that padding in grid layouts

Reply:
@bigint No, we don't have that padding in grid layouts
I think I should add that padding in grid layout.

Reply:
@bigint We don't have that padding in grid layouts

Reply:
@bigint The “sticky” value of the navbar position property is a mixture of the relative and fixed positioning (which adjusts the content by itself, the actual content will start after the navbar div/header). But with radix, sticky navbar doesn't work since it makes body overflow as hidden (, that's why if we will have to make navbar position to fixed. If we have made navbar position to fixed, then we will need to adjust the content by adding a padding top which creates that separation)
The padding in Grid Layouts is given by considering that the content will just after the Navbar (assuming its sticky position). But since we have changed the position from sticky to fixed, we will need to add this additional padding top.

Reply:
<img width="1512" alt="Screenshot 2023-12-30 at 1 58 30 PM (2)" src="https://github.com/heyxyz/hey/assets/68169950/691b77c5-35e5-4399-b834-ef6a3751b08f">
Please refer this screenshot and https://michaelmovsesov.com/articles/fix-css-position-sticky-not-working

Reply:
@bigint Got the actual issue, which is causing the behaviour of radix modal to deviate from its original behaviour.
https://github.com/heyxyz/hey/blame/dc27d01f3b899cf759279ca39c4bbb3bf1ecd4f5/apps/web/src/styles.css#L7
This line is causing issue, is there a need to apply overflow scroll y on whole html document, I don't think so because we have already used overflow tailwind classes to handle this and anyways applying classes directly to html element is also not good practice.
Do you have some opinions on it?
Will it be fine I remove this?

Reply:
@bigint Fixed this in latest commit!

Reply:
@bigint Any updates on this? Is now everything fine?

Reply:
@bigint Any Updates???